IntegrationTests/Services/AWSEC2IntegrationTests/IMDSCredentialsProviderTests.swift

```diff
@@ -0,0 +1,294 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+import XCTest
+import AWSIAM
+import AWSSTS
+import AWSCloudWatchLogs
+import AWSEC2
+import ClientRuntime
+
+class IMDSCredentialsProviderTests: XCTestCase {
+    private let region = "us-west-2"
+    private var ec2Client: EC2Client!
+    private var iamClient: IAMClient!
+    private var stsClient: STSClient!
+    private var cloudWatchLogClient: CloudWatchLogsClient!
+
+    // Poll amount maixmum with a 10 second wait between each attempt.
+    private var maxPollLogGroupCreation = 50
+    private var maxPollLogMessageFound = 10
+
+    // MARK: - SETUP & TEARDOWN
+
+    override func setUp() async throws {
+        ec2Client = try EC2Client(region: region)
+        iamClient = try IAMClient(region: region)
+        stsClient = try STSClient(region: region)
+        cloudWatchLogClient = try CloudWatchLogsClient(region: region)
+        try await launchEC2Instance()
+        // Wait until cloud watch log group is created by shell script passed to EC2 instance at launch.
+        try await awaitCloudWatchLogSetup()
+    }
+
+    // IAM role & instance profile and docker image in ECR private repo are kept for re-use.
+    override func tearDown() async throws {
+        _ = try await ec2Client.terminateInstances(input: TerminateInstancesInput(
+            instanceIds: [ec2InstanceID]
+        ))
+        _ = try await cloudWatchLogClient.deleteLogGroup(
+            input: DeleteLogGroupInput(logGroupName: cloudWatchLogGroupName)
+        )
+    }
+
+    // MARK: - TEST CASE
+
+    // Delete `x` to enable test.
+    func xtestIMDSCredentialsProvider() async throws {
```

Comment:
`xtest` is actually a common way to disable tests, not just in Swift tests but with other languages' test suites as well.  (Not XCTest AFAIK, but some test frameworks actually see the `x` before a test & list it in results as a disabled test.)

Searching `xtest` in the project should take you to disabled tests & nothing else, since it's not a word that's likely to appear anywhere but in a disabled test.